util/partition_integers

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env python
+# partition_integers -- Partitions integer range into even ranges
+##
+import sys
+
+M = int(sys.argv[1])
+N = int(sys.argv[2])
+
+k = float(M)/N
```

Comment:
Thanks for catching this.  I wasn't careful enough to include unit tests for this.  I'll make sure it produces error on such case.  Yes if we want more partitions than the number of elements, it'll end up with something like that.  I think it'll produce the a correct partition that covers all elements, right?